### PR TITLE
block form closing if it fails validation

### PR DESCRIPTION
**Pull request by yoshokatana**
* [trello ticket that prompted this](https://trello.com/c/3wPlH7Bm/55-bug-image-and-video-components-should-not-be-able-to-be-published-with-invalid-urls)
* forms will not close if they have blocking (e.g. native validation) errors
* trying to close form will trigger native validation messages
* other services that try and close forms will hit the validation as well

### CODE REVIEW

**Comment by cruzanmo** [comment]
`Firefox 42.0.0 (Mac OS X 10.10.0) ERROR`

File: controllers/overlay.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ module.exports = function () {
     close: function (e) {
       if (e.target === e.currentTarget) {
         // we clicked on the overlay itself
-        focus.unfocus();
+        focus.unfocus().catch(_.noop);
```
Comment (cruzanmo) [inline controllers/overlay.js]: This allows the form to close when it is invalid?

Comment (yoshokatana) [inline controllers/overlay.js]: No, that's a noop. It'll stay open, but not throw an error

Comment (cruzanmo) [inline controllers/overlay.js]: Oh, I see, that makes sense.

**Comment by cruzanmo** [comment]
+1

**Pull request merged**